dm/docs/RFCS/20220202_optimist_shard_mode_support_conflict_ddl.md

```diff
@@ -0,0 +1,155 @@
+
+A simple solution is to simply check whether the DDLs are the same, just as what we do for pessimistic DDL coordination. Coordination is completed when all sharding tables encounter the same conflicting DDL. This approach has the following problems.
+
+- If the sharding DDLs are not identical (e.g. different comment information), then the coordination will fail
+- As with pessimistic reconciliation, there may be problems with failover cases
+  - DDL1, DDL2 are executed on sharding table 1 and 2
+  - master coordination of DDL1 is complete
+  - worker1 executes DDL1 successfully, then tries to coordinate DDL2
+  - worker2 crashes while executing DDL1 (DDL and checkpoint are not atomic), restarts and recoordinates DDL1
+  - master receives DDL2 from worker1, DDL1 from worker2, an error will be reported
+- Since our optimistic coordination allows ***inconsistent but comparable*** initial table structures, we also encounter the following problems.
+  - initial structure tb1(a int, b int), tb2(a int, b varchar(4))
+  - tb1 execute `modify b varchar(4)`
+  - at this point, all the sub-tables are consistent, but the master only receives the DDL for tb1, so coordination fails
+
+### Comparison based on the schema before and after DDL execution
+
+Instead of considering specific DDLs, this approach determines whether all the sharding tables have been optimistic compatible based on the comparison of the schema before and after the DDL execution.
+
+#### Consistency condition for sharding tables
+
+Just to be clear, for a conflicting DDL that makes the table structure "big and small", suppose it makes table "-a, +b". When all the sharding tables have no `a` and have `b` at some point, then the DML synchronization of that sharding table can continue. e.g.
+
+- If `rename column a to b`, when all sharding tables `rename column a to b`, then there is no conflict
+- If `change a int to varchar`, when all sharding tables `change a int to varchar`, then there is no conflict
+- If `add a int not null`, when all sharding tables `add a int not null`, then there is no conflict
+- If table 1 `rename column a to b`, table 2 `drop column a, add column b`, then the downstream table structure has no `a`, has `b`, all the sharding tables can continue to synchronize DML, ***but the upstream and downstream data may be inconsistent***
+- If table 1 `change a int to varchar`, table 2 `drop column a, add column a varchar`, then all tables can continue to synchronize DML, ***but the upstream and downstream data may be inconsistent***
+- If table 1 `add a int not null`, `table 2 add a int not null default 0`, then all sharding tables can continue to synchronize DML.
+
+***For the above possible data inconsistency, similar to the user did a truncate operation on a column of a sharding table, unless the user redo the task, the upstream and downstream data must be inconsistent, and there is no way to avoid this situation.***
```

Comment:
```suggestion
***For the above possible data inconsistency problems, it's similar to the user did a truncate SQL on a column of a sharding table. Unless the user redo the task, the upstream and downstream data will always be inconsistent, and there is no way to avoid this situation currently.***
```